src/main/java/org/fenixedu/academic/ui/struts/action/phd/coordinator/publicProgram/PublicPhdProgramCandidacyProcessDA.java

```diff
@@ -165,6 +166,20 @@ public ActionForward manageFocusAreas(ActionMapping mapping, ActionForm form, Ht
         return mapping.findForward("manageFocusAreas");
     }
 
+    public ActionForward toggleFocusAreaAction(ActionMapping mapping, ActionForm form, HttpServletRequest request,
+            HttpServletResponse response) {
+
+        PhdProgramFocusArea focusArea = getDomainObject(request, "focusAreaId");
+        toggleFocusAreaActivation(focusArea);
+
+        return redirect("/candidacies/phdProgramCandidacyProcess.do?method=manageFocusAreas", request);
+    }
+
+    @Atomic
+    public void toggleFocusAreaActivation(PhdProgramFocusArea focusArea) {
+        focusArea.setActive(!focusArea.getActive());
+    }
```

Comment:
There is a better way without adding a new method. Something [like](https://github.com/FenixEdu/fenixedu-academic/blob/v5.4.2/src/main/java/org/fenixedu/academic/ui/struts/action/person/UpdateNicknameDA.java#L48-L52) this.